src/providers/viaMichelin/ViaMichelin.js

```diff
@@ -133,48 +139,56 @@ class ViaMichelinMap extends Map {
         }
     }
 
-    addMarkers(points) {
+    addMarkers(points, clusterIndex = 0, clusterConfig) {
         if (Object.prototype.toString.call(points) !== '[object Array]') {
             points = [points];
         }
 
         let markers = [];
         let options = {};
-        let activeInfoWindow;
 
         for(let i = 0; i < points.length; i++) {
             options = Object.assign({}, this.options.marker, points[i].options ? points[i].options : {});
-            activeInfoWindow = points[i].activeInfoWindow !== undefined ? points[i].activeInfoWindow : this.options.activeInfoWindow;
+
+            if (options.activeInfoWindow === undefined) {
+                options.activeInfoWindow = this.options.activeInfoWindow;
+            }
+
+            if (options.activeCluster === undefined) {
+                options.activeCluster = this.options.activeCluster;
+            }
 
             points[i].options = options;
-            points[i].activeInfoWindow = activeInfoWindow;
 
             if (this.map) {
-                const marker = new Marker(points[i], options, activeInfoWindow);
+                const marker = new Marker(points[i], options);
                 markers.push(marker);
                 this.markers.push(marker);
                 this.map.addLayer(marker);
             }
         }
-        this.setCluster();
+
+        this.setCluster(markers, clusterIndex, clusterConfig);
 
         return markers;
     }
 
-    setCluster() {
-        // Init the clustering if the option is set
-        if (this.options.activeCluster) {
-            if (this.cluster) {
-                // Reset all markers & clusters because ViaMichelin API does not allow it...
-                this.map.removeAllLayers();
-                this.cluster.clear();
-
-                // Redraw each markers
-                this.markers.forEach((marker) => {
-                    this.map.addLayer(marker);
-                });
+    setCluster(markers, clusterIndex, clusterConfig) {
+        if (this.map && this.options.activeCluster && clusterIndex !== false) {
```

Comment:
in fact it's the index of `this.markerClusterers`, or false if you don't want to clusterize your markers.
Maybe I should use `null` instead...